File: Data Structures & Algorithms/surrounded-regions/submission-0.py
class Solution:
    def solve(self, board: List[List[str]]) -> None:
        if not board:
            return
        m, n = len(board), len(board[0])

        def dfs(r, c):
            if 0 <= r < m and 0 <= c < n and board[r][c] == 'O':
                board[r][c] = 'T'
                for dr, dc in [(-1, 0), (1, 0), (0, -1), (0, 1)]:
                    dfs(r+dr, c+dc)
        
        for i in range(m):
            dfs(i, 0)
            dfs(i, n-1)
        for j in range(n):
            dfs(0, j)
            dfs(m-1, j)
        for i in range(m):
            for j in range(n):
                if board[i][j] == 'O':
                    board[i][j] = 'X'
                elif board[i][j] == 'T':
                    board[i][j] = 'O'

File: Data Structures & Algorithms/valid-sudoku/submission-0.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        HashSet<Character> row[] = new HashSet[9];
        HashSet<Character> col[] = new HashSet[9];
        HashSet<Character> box[] = new HashSet[9];

        for (int i = 0; i < 9; i++) {
            row[i] = new HashSet();
            col[i] = new HashSet();
            box[i] = new HashSet();
        }

        for (int r = 0; r < 9; r++) {
            for (int c = 0; c < 9; c++) {
                char value = board[r][c];
                int boxId = r / 3 * 3 + c / 3;

                if (value == '.') continue;
                if (row[r].contains(value) || col[c].contains(value) || box[boxId].contains(value)) {
                    return false;
                }
                row[r].add(value);
                col[c].add(value);
                box[boxId].add(value);
            }
        }
        return true;
    }
}
